concept/type_/type_manager.rs

```diff
@@ -1480,6 +1601,12 @@ impl TypeManager {
         owner: ObjectType<'static>,
         attribute: AttributeType<'static>,
     ) -> Result<(), ConceptWriteError> {
+        with_object_type!(owner, |type_| {
+            OperationTimeValidation::validate_can_modify_type(snapshot, type_)
+                .map_err(|source| ConceptWriteError::SchemaValidation { source })?;
+        });
+
+        // TODO: We have a commit-time check for overrides not being left hanging, but maybe we should clean it/reject here. We DO check that there are no subtypes for type deletion.
```

Comment:
Kinda don't want to overdo here, but it's an idea for the future once we touch typeql by hands (or I can do it right now if you want to)